site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+
+```
+metadata:
+  name: foo_route
+  namespace: store
+spec:
+  parentRefs:
+  - kind: Service
+    name: foo
+  rules:
+    matches:
+    - path:
+        value: "/bar"
+```
+
+The example above would drop all incoming traffic for HTTP paths other than `/bar` to the `foo` `Service`, following the existing spec to return HTTP 404 response codes for unmatched requests, and HTTP 500 response codes for requests excluded due to an [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter).
+
+### Allowed `Service` types
+
+Valid `Service` resources to be selected as a `parentRef` SHOULD provision one or more virtual IP(s) and DNS hostname(s) within the Kubernetes cluster, and include the default [`ClusterIP`](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types) `Service` type, with or without selectors.
+
+`Service` resource with [`type: NodePort`](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport) or [`type: LoadBalancer`](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) MUST NOT be allowed as `parentRefs` or `backendRefs`. It should isntead be preferred to expose services publicly through the North/South Gateway API interfaces.
+
+`Services` with [`type: ExternalName`](https://kubernetes.io/docs/concepts/services-networking/service/#externalname) SHOULD NOT be allowed as `parentRefs` or `backendRefs` due to [security concerns](https://github.com/kubernetes/kubernetes/issues/103675), although might eventually play some role in [configuring egress functionality](https://github.com/kubernetes-sigs/gateway-api/issues/1070).
+
+`Service` resources supported as `backendRefs` SHOULD be consistent with expectations for North/South Gateway API implementations, MUST have associated endpoints, and include:
+
+* `ClusterIP` `Services` with selectors
+* [Headless Services](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) with selectors. Headless services without selectors SHOULD NOT be supported due to the possibility of matching `Services` with `type: ExternalName`.
+* `StatefulSet` resources
+
+#### `Service` without selectors
+
+An alternate pattern additionally supported by this approach would be to target a `Service` without selectors as the `parentRef`. This could be a clean way to create a pure routing construct and abstract a logical frontend, as traffic would resolve to a `backendRef` `Service` with selectors defined on the `HTTPRoute`, or receive a 4xx/5xx error response if no matching path or valid backend was found.
+
+#### StatefulSet
+
+[`StatefulSet`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) resources MAY be supported as a `backendRef`, but specifics of this integration have not yet been clarified in the full Gateway API spec.
```

Comment:
I don't really understand the purpose of this, I think it would be safest to remove it unless there are well defined use cases we need to support.

Reply:
```suggestion
```
There's at least three separate threads suggesting to remove this, so I'll get rid of it.